crates/runtime/src/accelerated_table.rs

```diff
@@ -228,7 +241,14 @@ impl Builder {
         let (acceleration_refresh_mode, refresh_trigger) = match self.refresh.mode {
             RefreshMode::Append => {
                 if self.refresh.time_column.is_none() {
-                    (refresh::AccelerationRefreshMode::Append(None), None)
+                    // Get the append stream
+                    let Some(append_stream) = self.append_stream else {
+                        panic!("Append stream is required for `RefreshMode::Append` without time_column");
```

Comment:
Do we do validation upstream to ensure we can't hit this case? We shouldn't panic as the result of a user misconfiguring something, panics should only be for cases that should be impossible to hit. The panic in the changes stream below won't be hit because of validation that happens upstream.